### PR TITLE
vanchor governance

### LINKED ISSUE #192: [Task] Substrate Vanchor watcher for governance
## Overview
Similar to the regular `Anchor` watcher for governance,  we need to implement the same for `Vanchor`.

Vanchor Governance Flow
```
1. Go through every linked edges.
2. Construct AnchorUpdateProposal from the deposit information.
3. Check if the ProposalSigningBackend can sign it and if so we hand over the proposal.
4. Then we mark this deposit as processed.
```

vanchor watcher for [evm](https://github.com/webb-tools/relayer/blob/main/src/events_watcher/evm/vanchor_watcher.rs)

### PR DESCRIPTION
**Pull request by salman01zp**
## Task Checklist
- [x] Implement Vanchor Watcher
- [x] Update `Protocol-substrate` to  include `vanchor-handler` in runtime. [PR](url)
- [x] Update substrate metadata in webb-rs [PR](url)
- [x] Add tests for Vanchor Watcher

**Reference issue to close (if applicable)**

Closes #192 

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
- [x] Yes
- [ ] No

### CODE REVIEW

File: src/events_watcher/substrate/signature_bridge_watcher.rs [hunk 1]
```diff
@@ -219,6 +221,15 @@ where
                     match maybe_success {
                         Ok(events) => {
                             tracing::debug!(?events, "tx finalized",);
+                            tracing::event!(
+                                target: crate::probe::TARGET,
+                                tracing::Level::DEBUG,
+                                kind = %crate::probe::Kind::SignatureBridge,
+                                call = "executed_proposal_with_signature",
```
Comment (shekohex) [inline src/events_watcher/substrate/signature_bridge_watcher.rs:228]:
```suggestion
                                call = "execute_proposal_with_signature",
```

File: src/events_watcher/substrate/vanchor_watcher.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+use std::sync::Arc;
+use webb::substrate::protocol_substrate_runtime::api::v_anchor_bn254;
+use webb::substrate::scale::Encode;
+use webb::substrate::{protocol_substrate_runtime, subxt};
+use webb_proposals::substrate::AnchorUpdateProposal;
+
+/// Represents an Anchor Watcher which will use a configured signing backend for signing proposals.
+pub struct SubstrateVAnchorWatcher<'a, B> {
+    proposal_signing_backend: B,
+    linked_anchors: &'a [SubstrateLinkedAnchorConfig],
```
Comment (shekohex) [inline src/events_watcher/substrate/vanchor_watcher.rs:29]: Why do we need this with a lifetime? why not just a clone of the that as `Vec<SubstrateLinkedAnchorConfig>`?

File: tests/lib/webbRelayer.ts [hunk 3]
```diff
@@ -636,3 +636,40 @@ function parseRelayTxMessage(o: any): ParsedRelayerMessage {
     return { kind: 'unknown' };
   }
 }
+
+export function getChainIdType(chainID: number = 31337): number {
```
Comment (shekohex) [inline tests/lib/webbRelayer.ts:640]: Chain Id type for? Substrate? EVM? this function name does not say what type, besides it takes a predefined default parameter.

Please make the name chain specific like `getSubstrateTypedChainId` for example.

Also, I don't think this not being used?

Comment (shekohex) [reply tests/lib/webbRelayer.ts:640]: There is ResouceId somewhere in webbProposals.js that can be used to encode/create these kind of typed chainId btw. you should use them instead.

**Review by shekohex — CHANGES_REQUESTED** [review]

**Review by drewstone — APPROVED** [review]
LGTM

**Pull request merged**